### PR TITLE
Fix/noise units

### PR DESCRIPTION
**Pull request by flaviofontana**
Hi Simon 

I integrated your comments. Please have a look and let me know if I should change anything. Please double check the units on the imu bias random walk . 
if you agree I'd like to extend the viconpos_sensor_fix.yaml with the same comments. So that people following the tutorial will also have the comments available.

flavio

### CODE REVIEW

File: msf_core/cfg/MSF_Core.cfg [hunk 1]
```diff
@@ -28,11 +28,30 @@ gen = ParameterGenerator()
...
 # @todo Double check these ranges, defaults
 
+
+# The imu measurement model used in msf contains two types of sensor errors, a fast additive white noise and 
+# a slower varying sensor bias.  
+# See the following link for more information 
+# https://github.com/ethz-asl/kalibr/wiki/IMU-Noise-Model-and-Intrinsics
+# 
+# The white noise is characterized with the continous time noise spectral density. 
+# The noise spectral density is sometime also refered to as noise density.
```
Comment (nikolicj) [inline msf_core/cfg/MSF_Core.cfg]: referred

**Comment by simonlynen** [comment]
add to whitelist

File: msf_core/cfg/MSF_Core.cfg [hunk 2]
```diff
@@ -28,11 +28,30 @@ gen = ParameterGenerator()
 # @todo Think about levels. Setting most of these to STOP to guarantee atomicity.
 # @todo Double check these ranges, defaults
 
+
+# The imu measurement model used in msf contains two types of sensor errors, a fast additive white noise and
```
Comment (simonlynen) [inline msf_core/cfg/MSF_Core.cfg]: IMU

Comment (simonlynen) [inline msf_core/cfg/MSF_Core.cfg]: fast -> high frequency

File: msf_core/cfg/MSF_Core.cfg [hunk 3]
```diff
@@ -28,11 +28,30 @@ gen = ParameterGenerator()
...
+# a slower varying sensor bias.  
+# See the following link for more information 
+# https://github.com/ethz-asl/kalibr/wiki/IMU-Noise-Model-and-Intrinsics
+# 
+# The white noise is characterized with the continous time noise spectral density. 
+# The noise spectral density is sometime also refered to as noise density.
+# The units of the noise spectral density are:
+#  acc:  [m/s^2/sqrt(Hz)]
+#  gyro: [rad/s/sqrt(Hz)]
+# The noise spectral density can be found in the datasheet.
```
Comment (simonlynen) [inline msf_core/cfg/MSF_Core.cfg]: datasheet of the IMU.

File: msf_updates/pose_sensor_fix.yaml [hunk 4]
```diff
@@ -1,33 +1,72 @@
+data_playback: true             # set to true when playback, false on the real system
```
Comment (simonlynen) [inline msf_updates/pose_sensor_fix.yaml]: Sentensify comment: Capital letter and period.

File: msf_updates/pose_sensor_fix.yaml [hunk 5]
```diff
@@ -1,33 +1,72 @@
...
 
-data_playback: true
+##############################
+#########IMU PARAMETERS#######
+##############################
+# The imu measurement model used in msf contains two types of sensor errors, a fast additive white noise and
```
Comment (simonlynen) [inline msf_updates/pose_sensor_fix.yaml]: fast -> high frequency

File: msf_updates/pose_sensor_fix.yaml [hunk 6]
```diff
@@ -1,33 +1,72 @@
...
+##############################
+#########IMU PARAMETERS#######
+##############################
+# The imu measurement model used in msf contains two types of sensor errors, a fast additive white noise and 
+# a slower varying sensor bias.  
+# See the following link for more information 
+# https://github.com/ethz-asl/kalibr/wiki/IMU-Noise-Model-and-Intrinsics
+# 
+# The white noise is characterized with the continous time noise spectral density. 
+# The noise spectral density is sometime also refered to as noise density.
```
Comment (simonlynen) [inline msf_updates/pose_sensor_fix.yaml]: same as above

File: msf_updates/pose_sensor_fix.yaml (hunk 5 above)
Comment (simonlynen) [inline msf_updates/pose_sensor_fix.yaml]: IMU

File: msf_core/cfg/MSF_Core.cfg [hunk 7]
```diff
@@ -28,11 +28,30 @@ gen = ParameterGenerator()
...
-gen.add("core_noise_acc",         double_t, 0,                           "noise accelerations (std. dev)",           0.002,   	1.0e-4, 	0.5)
-gen.add("core_noise_accbias",     double_t, 0,                           "noise acceleration bias  (std. dev)",       5e-8, 	1.0e-8, 	0.1)
-gen.add("core_noise_gyr",         double_t, 0,                           "noise gyros (std. dev)",           0.0004,  	1.0e-4,   	0.5)
-gen.add("core_noise_gyrbias",     double_t, 0,                           "noise gyro biases  (std. dev)",       3e-6, 	1.0e-7, 	0.1)
+gen.add("core_noise_acc",         double_t, 0,                           "accelerations noise spectral density (nsd) [m/s^2/sqrt(Hz)]",       0.002,   	1.0e-4, 	0.5)
+gen.add("core_noise_accbias",     double_t, 0,                           "acceleration biases random walk [m/s^3/sqrt(Hz)]", 5e-8, 	1.0e-8, 	0.1)
+gen.add("core_noise_gyr",         double_t, 0,                           "gyros noise spectral density (nsd) [rad/s/sqrt(Hz)]",                 0.0004,  	1.0e-4,   	0.5)
+gen.add("core_noise_gyrbias",     double_t, 0,                           "gyro biases random walk [rad/s^2/sqrt(Hz)]",                 3e-6, 	1.0e-7, 	0.1)
```
Comment (nikolicj) [inline msf_core/cfg/MSF_Core.cfg]: That looks correct to me!

File: msf_updates/pose_sensor_fix.yaml [hunk 8]
```diff
@@ -1,33 +1,72 @@
...
+# The units of the noise spectral density are:
+#  acc:  [m/s^2/sqrt(Hz)]
+#  gyro: [rad/s/sqrt(Hz)]
+# The noise spectral density can be found in the datasheet.
```
Comment (simonlynen) [inline msf_updates/pose_sensor_fix.yaml]: datasheet of the IMU

File: msf_updates/pose_sensor_fix.yaml [hunk 9]
```diff
@@ -1,33 +1,72 @@
...
+# 
+# The variation of the bias is characterized as a random walk. 
+# Again see https://github.com/ethz-asl/kalibr/wiki/IMU-Noise-Model-and-Intrinsics for more information
+# The units of the random walk are:
+#  acc:  [m/s^3/sqrt(Hz)]
+#  gyro: [rad/s^2/sqrt(Hz)]
+ 
+####### ADIS
```
Comment (simonlynen) [inline msf_updates/pose_sensor_fix.yaml]: ADIS 16488

File: msf_updates/pose_sensor_fix.yaml [hunk 10]
```diff
@@ -1,33 +1,72 @@
...
+#core/core_fixed_bias: true
+#core/core_noise_gyrbias: 0.0     # for fixed bias we do not need process noise
+#core/core_noise_accbias: 0.0     # for fixed bias we do not need process noise
 
+#######################################
+#########Pose Sensor Parameters #######
+#######################################
+pose_sensor/pose_absolute_measurements: true
+pose_sensor/pose_measurement_world_sensor: false  # selects if sensor measures its position w.r.t. world (true, e.g. Vicon) or the position of the world coordinate system w.r.t. the sensor (false, e.g. ethzasl_ptam
+pose_sensor/pose_delay: 0.02                      # [s] delay of pose sensor w.r.t. imu
```
Comment (simonlynen) [inline msf_updates/pose_sensor_fix.yaml]: IMU

Comment (simonlynen) [inline msf_updates/pose_sensor_fix.yaml]: sentensify comment

File: msf_updates/pose_sensor_fix.yaml [hunk 11]
```diff
@@ -1,33 +1,72 @@
...
+
+# for the pose sensor noise levels use the std deviation the units are
```
Comment (simonlynen) [inline msf_updates/pose_sensor_fix.yaml]: Sentensify here and below

File: msf_updates/pose_sensor_fix.yaml [hunk 12]
```diff
@@ -1,33 +1,72 @@
...
+
+pose_sensor/pose_initial_scale: 1
 pose_sensor/pose_fixed_scale: false
-pose_sensor/pose_noise_scale: 0.0
-pose_sensor/pose_noise_p_wv: 0.0
-pose_sensor/pose_noise_q_wv: 0.0
-pose_sensor/pose_noise_q_ic: 0.0
+pose_sensor/pose_noise_scale: 0.0     
+
+# transformation between world frame (gravity aligned) and vision frame
```
Comment (simonlynen) [inline msf_updates/pose_sensor_fix.yaml]: Sentensify here and below

File: msf_updates/pose_sensor_fix.yaml [hunk 13]
```diff
@@ -1,33 +1,72 @@
...
+pose_sensor/pose_fixed_p_wv: false              
+pose_sensor/pose_noise_p_wv: 0.0                  
+pose_sensor/pose_fixed_q_wv: false
+pose_sensor/pose_noise_q_wv: 0.0                  
+
+# transformation between pose sensor and IMU
```
Comment (simonlynen) [inline msf_updates/pose_sensor_fix.yaml]: Transformation that expresses the position and orientation of the pose-sensor w.r.t. the IMU frame of reference, expressed in the IMU frame of reference.

**Comment by simonlynen** [comment]
@flaviofontana Thanks, I added some comments, otherwise this looks already much better than what we had before!

**Comment by flaviofontana** [comment]
@simonlynen I can make the changes, but how? Can i just commit again or do i need to make a new pull request?

**Comment by simonlynen** [comment]
you just push more commits to this.

File: msf_updates/pose_sensor_fix.yaml (hunk 9 above)
Comment (flaviofontana) [inline msf_updates/pose_sensor_fix.yaml]: @simonlynen  this is from your config file, I dont know where these values come from

Comment (simonlynen) [inline msf_updates/pose_sensor_fix.yaml]: 16448

File: msf_core/cfg/MSF_Core.cfg (hunk 1 above)
Comment (simonlynen) [inline msf_core/cfg/MSF_Core.cfg]: @flaviofontana bump

File: msf_updates/pose_sensor_fix.yaml (hunk 6 above)
Comment (simonlynen) [inline msf_updates/pose_sensor_fix.yaml]: bump

**Comment by flaviofontana** [comment]
@simonlynen  let me know if i forgot something

File: msf_updates/pose_sensor_fix.yaml [hunk 14]
```diff
@@ -1,33 +1,72 @@
...
-pose_sensor/pose_noise_q_ic: 0.0
+pose_sensor/pose_noise_scale: 0.0     
+
+# Transformation that expresses the position and orientation of the gravity aligned world frame w.r.t the vision/camera frame
+pose_sensor/pose_fixed_p_wv: false              
+pose_sensor/pose_noise_p_wv: 0.0                  
+pose_sensor/pose_fixed_q_wv: false
+pose_sensor/pose_noise_q_wv: 0.0                  
+
+# Transformation that expresses the position and orientation of the pose-sensor w.r.t. the IMU frame of reference, expressed in the IMU frame of reference.
```
Comment (simonlynen) [inline msf_updates/pose_sensor_fix.yaml]: break at 100c

File: msf_updates/pose_sensor_fix.yaml [hunk 15]
```diff
@@ -1,33 +1,72 @@
...
+
+pose_sensor/pose_initial_scale: 1
 pose_sensor/pose_fixed_scale: false
-pose_sensor/pose_noise_scale: 0.0
-pose_sensor/pose_noise_p_wv: 0.0
-pose_sensor/pose_noise_q_wv: 0.0
-pose_sensor/pose_noise_q_ic: 0.0
+pose_sensor/pose_noise_scale: 0.0     
+
+# Transformation that expresses the position and orientation of the gravity aligned world frame w.r.t the vision/camera frame
```
Comment (simonlynen) [inline msf_updates/pose_sensor_fix.yaml]: break at 100c

**Comment by simonlynen** [comment]
@flaviofontana two micro formatting changes, then this is good to go. Thanks a lot for the cleanup and documentation. This is much better now!

**Comment by flaviofontana** [comment]
@simonlynen what do you think about removing msf_updates/aslam_sensor_fix.yaml
then we only have one good config file.

**Comment by simonlynen** [comment]
@flaviofontana yes please remove this file.

@markusachtelik @burrimi FYI, this removes aslam_sensor_fix.yaml and merges the configuration files for ADIS and MPU

**Comment by burrimi** [comment]
@simonlynen This is no issue for us, we have our config files in a separate repo.

**Pull request merged**